packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,579 @@
+///                           itemCount: options.length,
+///                           itemBuilder: (BuildContext context, int index) {
+///                             final String option = options.elementAt(index);
+///                             return GestureDetector(
+///                               onTap: () {
+///                                 onSelected(option);
+///                               },
+///                               child: ListTile(
+///                                 title: Text(option),
+///                               ),
+///                             );
+///                           },
+///                         ),
+///                       ),
+///                     ),
+///                   );
+///                 },
+///               ),
+///               ElevatedButton(
+///                 onPressed: () {
+///                   FocusScope.of(context).requestFocus(new FocusNode());
+///                   if (!_formKey.currentState.validate()) {
+///                     return;
+///                   }
+///                   showDialog<void>(
+///                     context: context,
+///                     builder: (BuildContext context) {
+///                       return AlertDialog(
+///                         title: Text('Successfully submitted'),
+///                         content: SingleChildScrollView(
+///                           child: ListBody(
+///                             children: <Widget>[
+///                               Text('DropdownButtonFormField: "$_dropdownValue"'),
+///                               Text('TextFormField: "${_textEditingController.text}"'),
+///                               Text('AutocompleteCore: "$_autocompleteSelection"'),
+///                             ],
+///                           ),
+///                         ),
+///                         actions: <Widget>[
+///                           TextButton(
+///                             child: Text('Ok'),
+///                             onPressed: () {
+///                               Navigator.of(context).pop();
+///                             },
+///                           ),
+///                         ],
+///                       );
+///                     },
+///                   );
+///                 },
+///                 child: Text('Submit'),
+///               ),
+///             ],
+///           ),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+class AutocompleteCore<T extends Object> extends StatefulWidget {
+  /// Create an instance of AutocompleteCore.
+  ///
+  /// [fieldViewBuilder] and [optionsViewBuilder] must not be null.
+  const AutocompleteCore({
+    required this.fieldViewBuilder,
+    required this.optionsViewBuilder,
+    required this.optionsBuilder,
+    this.displayStringForOption = _defaultStringForOption,
+    this.onSelected,
+  }) : assert(fieldViewBuilder != null),
```

Comment:
Are these assertions needed with NNBD?

Reply:
Looks like we still need these for a while? https://github.com/dart-lang/language/issues/1018.

Reply:
Ah good to know.  I'll keep these in then, and I'll make sure I include them for parameters like this.